standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/PartitionManagementTask.java

```diff
@@ -162,6 +164,10 @@ public void run() {
         setupMsckPathInvalidation();
         Configuration msckConf = Msck.getMsckConf(conf);
         for (Table table : candidateTables) {
+          if (MetaStoreUtils.isDbBeingFailedOver(msc.getDatabase(table.getCatName(), table.getDbName()))) {
```

Comment:
This is going to be costly. One HMS call per table. Can we maintain a cache?

Reply:
Might not be doable. Say somehow we cached db failover property and after this repl.failover.enbaled prop is set to true for that db. Now, how will we figure out that cached data is outdated and needs to be fetched again?

Reply:
Cache can be short lived and we can rebuild it every time. Then also, we will have it at one call per db at max.